Flaky test: TestContainerConsumedStats on Windows CRI Integration Test

### Description

Frequent CRI Integration test failures at test **TestContainerConsumedStats** on Windows platform(3 out of latest 6 CI builds) were seen on [main branch](https://github.com/containerd/containerd/actions/workflows/ci.yml?query=branch%3Amain).   [1](https://github.com/containerd/containerd/actions/runs/3858118554/jobs/6576376374), [2](https://github.com/containerd/containerd/actions/runs/3850312293/jobs/6560386269), [3](https://github.com/containerd/containerd/actions/runs/3844087935/jobs/6546937525). 
```
=== RUN   TestContainerConsumedStats
    container_stats_test.go:76: Create a pod config and run sandbox container
    main_test.go:672: Pull test image "registry.k8s.io/e2e-test-images/resource-consumer:1.10"
    container_stats_test.go:82: Create a container config and run container in a pod
    container_stats_test.go:99: Fetch initial stats for container
    container_stats_test.go:113: Initial container memory consumption is 41.117188 MB. Consume 100 MB and expect the reported stats to increase accordingly
E0106 20:39:21.086023    2732 remote_runtime.go:412] ExecSync 8d56f2e540bc6452b47fe46fccdc50c740cbc2ba02358ea[286](https://github.com/containerd/containerd/actions/runs/3858118554/jobs/6576376374#step:22:287)ac4abb680ee94a 'testlimit.exe -accepteula -d 25 -c 4' from runtime service failed: rpc error: code = DeadlineExceeded desc = failed to exec in container: timeout 30s exceeded: context deadline exceeded
    container_stats_test.go:129: 
        	Error Trace:	D:\a\containerd\containerd\src\github.com\containerd\containerd\container_stats_test.go:129
        	Error:      	Received unexpected error:
        	            	timeout exceeded
        	Test:       	TestContainerConsumedStats
--- FAIL: TestContainerConsumedStats (38.28s)
```


Seems the random failure was introduced in #7892 which was targeted for Linux.  Specifically this [change](https://github.com/containerd/containerd/pull/7892/commits/76d68b080ef69407129a5925ae5b97f1c9010dea#diff-2885e285816541779f8967567923949ba6195e3a517479cef9813c813ac63d1dL106).
 
Without further research, I think have a check of OS using goruntime.GOOS and reapply the previous method of test to Windows should be sufficient to fix. Raised PR #7935. 

@AkihiroSuda  Any suggestions? 

### Steps to reproduce the issue

Random.
Trigger serval builds, wait and see.
Or check the latest failed runs on [main branch](https://github.com/containerd/containerd/actions/workflows/ci.yml?query=branch%3Amain).

### Describe the results you received and expected

Random failures on CRI integration test TestContainerConsumedStats on Windows were seen after merging into main who previously passed the build.

I expect reliable and consistent test results.

### What version of containerd are you using?

HEAD of main, commit c0c3546de4caf462e4fafea2071a25208aa04346

### Any other relevant information

_No response_

### Show configuration if it is related to CRI plugin.

_No response_

--- a/integration/container_stats_test.go
+++ b/integration/container_stats_test.go
@@ -103,8 +103,14 @@ func TestContainerConsumedStats(t *testing.T) {
 		if err != nil {
 			return false, err
 		}
-		if s.GetWritableLayer().GetTimestamp() > 0 {
-			return true, nil
+		if goruntime.GOOS == "windows" {
+			if s.GetMemory().GetWorkingSetBytes().GetValue() > 0 {
+				return true, nil
+			}
+		} else {
+			if s.GetWritableLayer().GetTimestamp() > 0 {
+				return true, nil
+			}
 		}
 		return false, nil
 	}, time.Second, 30*time.Second))

